### DISCUSSION TITLE
toga support subprocess?

### DISCUSSION THREAD

**Discussion by Jzhenli** [beeware/toga#2793] [Q&A]
Hi, I try to run toga with huey (a lightweight task queue lib), and start the huey_consumer via subprocess in the toga startup function, it seems work when run briefcase dev, but after briefcase package will raise FileNotFoundError. Below is my example code,
[hueytest.zip](https://github.com/user-attachments/files/16780560/hueytest.zip)

BTW, my pc os is win11.

**Comment by freakboy3742** [comment]
*Toga* doesn't provide any support for subprocesses - Toga is a GUI toolkit. 

Any subprocess support is provided by the underlying Python libraries; and in the case of Briefcase, can be influenced by the packaging process.

If you have a subprocess that is working fine under `briefcase dev`, but is raising a `FileNotFoundError` under a packaged app, the likely cause is that you're using a relative path to identify your subprocess binary. If you're explicitly naming an executable as part of a subprocess call, you need to provide a fully qualified path when naming that binary. You can use toga's [`paths`](https://toga.readthedocs.io/en/stable/reference/api/app.html#toga.App.paths) attribute to build an absolute path.

There are also restrictions on some platforms. When running under `briefcase dev`, `sys.executable` is a Python interpreter. When running on desktop platforms, `sys.executable` becomes the stub binary that starts your app - and that binary *isn't* a standard Python interpreter. This inhibits some forms of multiprocessing, as you can't run `sys.executable` to start a new Python process.

On Mobile, you can't use subprocesses *at all*, because those platforms have a different process model.
